aptos-move/framework/src/built_package.rs

```diff
@@ -39,6 +39,9 @@ pub const UPGRADE_POLICY_CUSTOM_FIELD: &str = "upgrade_policy";
 /// Represents a set of options for building artifacts from Move.
 #[derive(Debug, Clone, Parser, Serialize, Deserialize)]
 pub struct BuildOptions {
+    /// Dev mode to use [dev-addresses]
```

Comment:
This doesn't really explain what this flag does, can you explain this in way more detail? My assumption is when you set this, it makes the compiler take into account dev-addresses?

Reply:
`dev-addresses` are never used unless you set this flag, and it's always been set to false on the compiler.